File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.swift
class Solution {
	func countConsistentStrings(_ allowed: String, _ words: [String]) -> Int {
		var allowedSet: Set<Character> = []
		for char in allowed {
			allowedSet.insert(char)
		}
		
		var result = 0
		
		for word in words {
	
			var isConsistent = true
			for char in word {
				if !allowedSet.contains(char) {
					isConsistent = false
					break
				}
			}
			
			if isConsistent {
				result += 1
			}
		}
		
		return result
	}
}

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.cs
public class Solution
{
  public int CountConsistentStrings(string allowed, string[] words)
  {
    int result = 0;
    HashSet<char> set = new();

    foreach (char c in allowed)
    {
      set.Add(c);
    }

    foreach (string word in words)
    {
      bool isValid = true;

      foreach (char c in word)
      {
        if (!set.Contains(c))
        {
          isValid = false;
          break;
        }
      }

      if (isValid)
      {
        result++;
      }
    }

    return result;
  }
}